### PR TITLE
Add Test case repository and start using it in flatdata/cpp/rust generators

### PR DESCRIPTION
**Pull request by VeaaC**
For #96

### CODE REVIEW

**Comment by VeaaC** [comment]
Waiting for #98 to add rust tests

File: flatdata-py/generator/tests/generators/schemas.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    result = list()
+    basedir = os.path.dirname(__file__)
+    test_dir = os.path.normpath(os.path.join(basedir, '..', '..', '..','..', 'test_cases'))
+    for path, subdirs, files in os.walk(test_dir):
+        for name in files:
+            if os.path.splitext(name)[1] == '.flatdata':
+                relpath = os.path.relpath(path, test_dir)
+                test_relpath = os.path.join(relpath, name)
+                expecation_path = os.path.join(basedir,
+                                               generator + '_expectations',
```
Comment (boxdot) [inline flatdata-py/generator/tests/generators/schemas.py]: @gferon: Could we apply some formatting tool to this file?

File: flatdata-py/generator/tests/generators/schemas.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+                                               os.path.splitext(test_relpath)[0]+'.' + extension)
+                result.append((os.path.join(path, name), expecation_path))
+    assert len(result) == 23, "Did not find expected number of test cases in " + test_dir
```
Comment (boxdot) [inline flatdata-py/generator/tests/generators/schemas.py]: I guess 23 might be a constant at the beginning of the file to make it easily updateble.

File: flatdata-py/generator/tests/generators/schemas.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import os
+
+def schemas_and_expectations(generator, extension):
```
Comment (boxdot) [inline flatdata-py/generator/tests/generators/schemas.py]: Could we add a little bit of documentation what are `generator` and `extension` and what is the return type. Unfortunately, it is hard to read python without static types.

File: flatdata-py/generator/tests/generators/schemas.py (hunk 1 above)
Comment (VeaaC) [reply flatdata-py/generator/tests/generators/schemas.py]: My auto-formatting plugin fave up in this file... I tried to move some expression into variable to improve a bit

File: flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    8,
+    (f, set_f, u64, u64, 0, 64));
+
+define_struct!(
+    I64,
+    RefI64,
+    RefMutI64,
+    schema::structs::I64,
+    8,
+    (f, set_f, i64, i64, 0, 64));
```
Comment (boxdot) [inline flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs]: Just a comment: It looks like we are not generating endline sometimes.

File: flatdata-py/generator/tests/generators/schemas.py (hunk 3 above)
Comment (VeaaC) [reply flatdata-py/generator/tests/generators/schemas.py]: Done

File: flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs (hunk 4 above)
Comment (VeaaC) [reply flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs]: Yeah, our newlines are a bit messed up in the template... sometimes too many, sometimes not enough for the last element of a loop, etc.

I tried to fix for 1h or so, but eventually gave up for now. Will have to do so systematically at some point.

File: flatdata-py/generator/tests/generators/schemas.py (hunk 2 above)
Comment (VeaaC) [reply flatdata-py/generator/tests/generators/schemas.py]: Done

File: flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs (hunk 4 above)
Comment (boxdot) [reply flatdata-py/generator/tests/generators/rust_expectations/structs/integers.rs]: No worries, was just a comment.

**Comment by VeaaC** [comment]
Added Rust tests as well and thus this depends on #98 now (too many changes were in that one, so I had to pull it in)

**Comment by boxdot** [comment]
> Added Rust tests as well and thus this depends on #98 now (too many changes were in that one, so I had to pull it in)

Since I was already fine with #98, I think this is fine. This review is in any case a most review, so, from my side we are fine to go. @gferon: Any objections?

**Pull request merged**